### ISSUE TITLE
BUG REPORT: Scrollbar Flickering Issue when width of columns matches / reaches near the provided grid parent width

### ISSUE THREAD

**Issue by nikhil1793** [ag-grid/ag-grid#8226]
```
[] bug report => Scrollbar Flickering Issue when width of columns matches / reaches near the provided grid parent width. 
```

**Providing a Reproducible Scenario**
Accepted reproducible scenarios are
- Scrollbar Flickering Issue when width of columns matches / reaches near the provided grid parent width. 
- It continue changing the overflow-y of `div with .ag-floating-top class` from hidden to scroll and then scroll to hidden infinite times.
- For reference use this plunker link `https://plnkr.co/edit/wjnaP5L5RQBmGBmJ?open=app.component.ts` [link](https://plnkr.co/edit/wjnaP5L5RQBmGBmJ?open=app.component.ts)

If reporting a bug make sure to state.

Current behaviour.
Expected behaviour. If possible back this up with our docs/examples if possible

**Current behavior**
  It continue changing the overflow-y of `div with .ag-floating-top class` from hidden to scroll and then scroll to hidden infinite times.
  End ups in scroll flickering issue infinitely.

**Expected behavior**
It should adjust the column within the grid and horizontal scrollbar should not be visible if all cumulative width of columns is less than the provided total width of grid parent div.

**Please tell us about your environment:**
Operating system : Windows 
IDE: VS Code 
package manager: NPM
HTTP server: Angular dev server

* **AG Grid version:** 31.3.2

* **Browser:** Chrome

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

https://github.com/ag-grid/ag-grid/assets/12994269/bae4faab-cddd-4bd4-a921-b76791ed566a

**Comment by AG-Zoheil** [comment]
Hello 

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference: **AG-12044**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be in the next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by mehakbansal28** [comment]
Hi @AG-Zoheil ,

I am also facing the same issue and using ag grid latest version i.e 32.1.0 . I can see that bug AG-12044 is still in opened state. Could you please confirm when it is expected to be fixed .
Thanks

**Comment by antoniopinarella** [comment]
same happens on ^30.2.0
soon our app will be upgraded to latest angular(from 16), and i hoped that aggrid will solve this in latest release.

**Comment by mehakbansal28** [comment]
Hi @AG-Zoheil , we raised this issue long time back. Could you please tell when it is expected to be get fixed.
Thanks

**Comment by AG-Zoheil** [comment]
Hi all,

thank you for your comments. This has been resolved and will be in our next release.

Kind regards,
Zoheil

**Comment by ggrablewski** [comment]
> Hi all,
> 
> thank you for your comments. This has been resolved and will be in our next release.
> 
> Kind regards, Zoheil

There was a release 32.3.0 on Oct 22, but I do not see AG-12044 in the changelog. Also not in any earlier release...

**Comment by AG-Zoheil** [comment]
Hi @ggrablewski,

As this fixed contained a breaking change it was not released in 32.3.0. It will be released in our next major version which will be at the beginning of December.

Thanks!
